### PR TITLE
refactor: community chat page

### PR DESCRIPTION
**Pull request by Paulmburu**
Screenshot_1636372571
Screenshot_1636372575

### CODE REVIEW

File: lib/presentation/community/chat_screen/pages/community_chat_screen_page.dart [hunk 1]
```diff
@@ -20,13 +18,20 @@ import 'package:myafyahub/presentation/core/widgets/app_bar/custom_app_bar.dart'
...
 
+  @override
+  _CommunityChatScreenPageState createState() =>
+      _CommunityChatScreenPageState();
+}
+
+class _CommunityChatScreenPageState extends State<CommunityChatScreenPage> {
   @override
   Widget build(BuildContext context) {
+    String newMessage = 'New Message';
```
Comment (KellyEgesa) [inline lib/presentation/community/chat_screen/pages/community_chat_screen_page.dart:34]: Extract this to app strings

File: test/widgets_tests/presentation/community/chat_screen/pages/community_chat_screen_page_test.dart [hunk 2]
```diff
@@ -39,16 +39,19 @@ void main() {
       expect(find.byType(ListView), findsOneWidget);
       expect(find.byType(MessageInput), findsOneWidget);
 
-      await tester.tap(find.byKey(moreAppBarKey));
-      await tester.pumpAndSettle();
-      expect(find.byType(ScaffoldMessenger), findsWidgets);
+      // await tester.tap(find.byKey(moreAppBarKey));
```
Comment (johnngugi) [inline test/widgets_tests/presentation/community/chat_screen/pages/community_chat_screen_page_test.dart:42]: Dead code

File: lib/presentation/community/chat_screen/pages/community_chat_screen_page.dart [hunk 3]
```diff
@@ -85,49 +90,51 @@ class CommunityChatScreenPage extends StatelessWidget {
...
             Align(
               alignment: Alignment.bottomCenter,
               child: MessageInput(
                 onChanged: (String value) {
-                  ScaffoldMessenger.of(context)
-                      .showSnackBar(snackbar(content: value));
+                  newMessage = value;
+                },
+                onTap: () {
+                  ScaffoldMessenger.of(context).showSnackBar(
```
Comment (KellyEgesa) [inline lib/presentation/community/chat_screen/pages/community_chat_screen_page.dart:133]:
```suggestion
                ScaffoldMessenger.of(context)
                        ..hideCurrentSnackBar()
                        ..showSnackBar(
```

To ensure that any existing snackbar is closed first

File: lib/presentation/router/router_generator.dart [hunk 4]
```diff
@@ -75,8 +75,17 @@ class RouteGenerator {
         );
 
       case BWRoutes.phoneLogin:
-        return MaterialPageRoute<LoginPage>(
-          builder: (_) => LoginPage(),
+        return MaterialPageRoute<CommunityChatScreenPage>(
+          builder: (_) => const CommunityChatScreenPage(
+            communityChatData: CommunityListItem(
+              title: 'My Afya Guide',
```
Comment (KellyEgesa) [inline lib/presentation/router/router_generator.dart]: Extract strings to app strings

File: lib/presentation/community/chat_screen/widgets/more_menu_drop_down.dart [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+      shape: const RoundedRectangleBorder(
+          borderRadius: BorderRadius.all(Radius.circular(10.0))),
+      key: popupMenuButtonKey,
+      onSelected: (int result) async {
+        switch (result) {
+          case 1:
+
+            /// Notify user
+            ScaffoldMessenger.of(context).showSnackBar(const SnackBar(
+              content: Text('Exit Group $comingSoonText'),
```
Comment (KellyEgesa) [inline lib/presentation/community/chat_screen/widgets/more_menu_drop_down.dart:29]: Extract app strings

File: lib/presentation/community/chat_screen/widgets/more_menu_drop_down.dart [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+            ));
+            break;
+          case 2:
+            ScaffoldMessenger.of(context).showSnackBar(const SnackBar(
+              content: Text('Group Info  $comingSoonText'),
```
Comment (KellyEgesa) [inline lib/presentation/community/chat_screen/widgets/more_menu_drop_down.dart:34]: Extract app strings

File: test/widgets_tests/presentation/community/chat_screen/widgets/message_input_test.dart [hunk 7]
```diff
@@ -22,6 +21,7 @@ void main() {
     });
 
     testWidgets('renders correctly', (WidgetTester tester) async {
+      String message = 'test';
```
Comment (KellyEgesa) [inline test/widgets_tests/presentation/community/chat_screen/widgets/message_input_test.dart:24]: Extract app strings

**Review by KellyEgesa — APPROVED** [review]

**Review by KellyEgesa — APPROVED** [review]

**Pull request merged**